module/documents/item.mjs

```diff
@@ -470,6 +470,48 @@ export default class Item5e extends Item {
 
   /* -------------------------------------------- */
 
+  /**
+   * Determine an item's proficiency level based on its parent actor's proficiencies.
+   * @protected
+   */
+  _prepareProficiency() {
```

Comment:
Since this is a new method, it might be a good time to start moving preparation methods into the system data models.

We could add a `prepareProficiency` method, possibly in a new `ProficiencyTemplate` or within the existing `EquippableTemplate` (though that is also inherited by some types without proficiency), that is called by `prepareFinalAttributes` as `this.system.prepareProficiency?.()`.

Then, similar to how we are handling the `abilityMod` getter, there could be a method for getting type-specific multipliers that certain types can override. So `SpellData#_proficiencyMultiplier` would always return `1`, `ToolData#_proficiencyMultiplier` would have the tool-specific code, etc.

Reply:
I was able to utilise a `_proficiencyMultiplier` getter, which definitely felt a lot better as it was able to remove some ugly ternaries and clean up the logic in `prepareProficiency`. I wasn't able to successfully move `prepareProficiency` out of `Item5e` and into the Data Models though. I couldn't find a good place for it: The main outlier was spells. I think a `ProficiencyTemplate` that defined the `proficient` field, and had a `prepareProficiency` method would be a good way to do it. But that would mean adding the `proficient` field to spells, which I was reluctant to do.

Let me know if you have any alternate suggestions, or if you think we should reconsider putting the `proficient` field on spells.

Reply:
Could this be part of data preparation on the sub-type itself rather than preparation that is managed by the Item? We are assigning the `Proficiency` instance to `this.system`. Could we just shift this to sub-type prearation and instead assign like:
```js
this.prof = new Proficiency(this.parent.system.attributes.prof, this.proficiencyMultiplier);
```
Or we have to do it here because of order-of-operations?

Reply:
Ideally, yes, but there's not a good place to put such preparation at the moment. `EquipmentData`, `WeaponData`, `ToolData`, and `SpellData` do not share a common ancestor (aside from the base `SystemDataModel`), so it would have to be duplicated for each of them.